In [1]:
#Importing all the required libraries
import selenium
from selenium import webdriver
from selenium.webdriver import Chrome
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.options import Options
from selenium.common.exceptions import NoSuchElementException
from selenium.webdriver.support.ui import Select
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import pandas as pd
import time


from bs4 import BeautifulSoup
import requests
import urllib.request
import sys
import os

###### Assignment 4.1:
    Scrape the details of most viewed videos on YouTube from Wikipedia:
Url= https://en.wikipedia.org/wiki/List_of_most-viewed_YouTube_videos
 You need to find following details:
 
A) Rank
B) Name
C) Artist
D) Upload date
E) Views

In [5]:
#Activating the Chrome Browser
driver=webdriver.Chrome('C:/Users/ak38/Desktop/FlipDataStore/chromedriver')

#Opening the URL
driver.get("https://en.wikipedia.org/wiki/List_of_most-viewed_YouTube_videos")      

# to maximize the browser window
driver.maximize_window()

In [7]:
#Extracting the table content from "Top 30 most-viewed YouTube videos"
videos=driver.find_elements_by_xpath('//table[@class="wikitable sortable jquery-tablesorter"][1]/tbody/tr')

len(videos)

30

In [49]:
#Creating the lists to store the extracted data
vrank=[]
vname=[]
vuploader=[]
vdate=[]
vviews=[]

#Fetching the ranks from the "Top 30 most-viewed YouTube videos" table
rank=driver.find_elements_by_xpath('//table[@class="wikitable sortable jquery-tablesorter"][1]/tbody/tr/td[1]')
for r in rank:
    vrank.append(r.text.replace(".",""))

#Fetching the video names from the "Top 30 most-viewed YouTube videos" table
name=driver.find_elements_by_xpath('//table[@class="wikitable sortable jquery-tablesorter"][1]/tbody/tr/td[2]')
for n in name:
    vname.append(n.text.replace('"','').split("[")[0])

#Fetching the video uploader/artist name from the "Top 30 most-viewed YouTube videos" table
uploader=driver.find_elements_by_xpath('//table[@class="wikitable sortable jquery-tablesorter"][1]/tbody/tr/td[3]')
for u in uploader:
    vuploader.append(u.text)

#Fetching the upload date from the "Top 30 most-viewed YouTube videos" table
date=driver.find_elements_by_xpath('//table[@class="wikitable sortable jquery-tablesorter"][1]/tbody/tr/td[5]')
for d in date:
    vdate.append(d.text)

#Fetching the no.of views(in billions) from the "Top 30 most-viewed YouTube videos" table
views=driver.find_elements_by_xpath('//table[@class="wikitable sortable jquery-tablesorter"][1]/tbody/tr/td[4]')
for v in views:
    vviews.append(v.text)


In [54]:
#Creating a DataFrame
data=pd.DataFrame({})
#Assigning all the lists to the dictionary keys
data['Rank']=vrank
data['Name']=vname
data['Artist']=vuploader
data['Upload_Date']=vdate
data['Views (in billions)']=vviews
#viewing the final dataset
data

Rank                                       Name  \
0     1                           Baby Shark Dance   
1     2                                  Despacito   
2     3                       Johny Johny Yes Papa   
3     4                               Shape of You   
4     5                              See You Again   
5     6   Masha and the Bear – Recipe for Disaster   
6     7                                Uptown Funk   
7     8  Learning Colors – Colorful Eggs on a Farm   
8     9                                  Bath Song   
9    10                              Gangnam Style   
10   11                Phonics Song with Two Words   
11   12                                      Sugar   
12   13                                      Sorry   
13   14                             Dame Tu Cosita   
14   15                                       Roar   
15   16                             Counting Stars   
16   17                          Thinking Out Loud   
17   18                                 Dark Horse   
18   19                                      Faded   
19   20                          Wheels on the Bus   
20   21                               Shake It Off   
21   22                                    Lean On   
22   23                             Girls Like You   
23   24                                   Bailando   
24   25                                 Let Her Go   
25   26                                   Mi Gente   
26   27                                    Perfect   
27   28                                      Hello   
28   29           Waka Waka (This Time for Africa)   
29   30                                     Axel F   

                            Artist        Upload_Date Views (in billions)  
0   Pinkfong Kids' Songs & Stories      June 17, 2016                8.74  
1                       Luis Fonsi   January 12, 2017                7.39  
2                      LooLoo Kids    October 8, 2016                5.43  
3                       Ed Sheeran   January 30, 2017                5.34  
4                      Wiz Khalifa      April 6, 2015                5.13  
5                       Get Movies   January 31, 2012                4.44  
6                      Mark Ronson  November 19, 2014                4.20  
7                      Miroshka TV  February 27, 2018                4.14  
8       Cocomelon – Nursery Rhymes        May 2, 2018                4.12  
9                              Psy      July 15, 2012                4.08  
10                       ChuChu TV      March 6, 2014                3.91  
11                        Maroon 5   January 14, 2015                3.48  
12                   Justin Bieber   October 22, 2015                3.44  
13                       El Chombo      April 5, 2018                3.38  
14                      Katy Perry  September 5, 2013                3.36  
15                     OneRepublic       May 31, 2013                3.31  
16                      Ed Sheeran    October 7, 2014                3.27  
17                      Katy Perry  February 20, 2014                3.08  
18                     Alan Walker   December 3, 2015                3.08  
19      Cocomelon – Nursery Rhymes       May 24, 2018                3.07  
20                    Taylor Swift    August 18, 2014                3.06  
21                     Major Lazer     March 22, 2015                3.05  
22                        Maroon 5       May 31, 2018                3.04  
23                Enrique Iglesias     April 11, 2014                3.04  
24                       Passenger      July 25, 2012                3.00  
25                        J Balvin      June 29, 2017                2.93  
26                      Ed Sheeran   November 9, 2017                2.86  
27                           Adele   October 22, 2015                2.84  
28                         Shakira       June 4, 2010                2.84  
29                      Crazy Frog      June 16,

###### Assignment 4.2:
    Scrape the details team India’s international fixtures from bcci.tv.
Url = https://www.bcci.tv/.
You need to find following details:

A) Match title (I.e. 1st ODI)
B) Series
C) Place
D) Date
E) Time

Note: - From bcci.tv home page you have reach to the international fixture page through code.

In [129]:
#Activating the Chrome Browser
driver=webdriver.Chrome('C:/Users/ak38/Desktop/FlipDataStore/chromedriver')

#Opening the URL
driver.get("https://www.bcci.tv/")      

# to maximize the browser window
driver.maximize_window()

In [130]:
#click on the International Tab to see the drop down menu
international=driver.find_element_by_xpath('//div[@class="navigation__drop-down drop-down drop-down--reveal-on-hover"][1]')
time.sleep(2)
international.click() #Click
time.sleep(2)

In [ ]:
#Clicking on the Fixtures button from the dropdown menu
fixtures_tab=driver.find_element_by_xpath('//div[@class="navigation__drop-down drop-down drop-down--reveal-on-hover"]/div/ul/li[1]/a').click()

In [222]:
#Creating the lists to store the table contents in the form of list
title=[]
series=[]
place=[]
date=[]
month=[]
time=[]

#Master element from where we can extract all the data contents
fixtures=driver.find_elements_by_xpath('//div[@class="js-list"]/a')

#Iterating through the elements to extract the desired data
for fix in fixtures:
    #Fetching the title from the browser
    title.append(fix.find_element_by_xpath('div[2]/div[2]/p/strong').text)
    #Fetching the series from the browser
    series.append(fix.find_element_by_xpath('div[2]/div[1]/span').text)
    #Fetching the place from the browser
    place.append(fix.find_element_by_xpath('div[2]/div[2]/p/span').text)
    #Fetching the time of the event from the browser
    time.append(fix.find_element_by_xpath('div[1]/div/div/span[2]').text)
    #Fetching the date of the event from the browser
    date.append(fix.find_element_by_xpath('div/div/span').text)
    #Fetching the month of the event from the browser
    month.append(fix.find_element_by_xpath('div/div/div/span').text)
    
#checking the lenth of the lists to verify
len(title), len(series), len(place), len(date), len(date), len(month), len(time)

(12, 12, 12, 12, 12, 12, 12)

In [229]:
#Creating a DataFrame
data=pd.DataFrame({})
#Assigning all the lists to the dictionary keys
data['Title']=title
data['Series']=series
data['Place']=place
data['date']=date
data['month']=month
data['Time']=time
#viewing the final dataset
data

Title Series                         Place date      month       Time
0      Final   TEST   The Ageas Bowl, Southampton   18       JUNE  15:00 IST
1    1st ODI    ODI  R Premadasa Stadium, Colombo   13       JULY  14:30 IST
2    2nd ODI    ODI  R Premadasa Stadium, Colombo   16       JULY  14:30 IST
3    3rd ODI    ODI  R Premadasa Stadium, Colombo   18       JULY  14:30 IST
4   1st T20I   T20I  R Premadasa Stadium, Colombo   21       JULY  19:00 IST
5   2nd T20I   T20I  R Premadasa Stadium, Colombo   23       JULY  19:00 IST
6   3rd T20I   T20I  R Premadasa Stadium, Colombo   25       JULY  19:00 IST
7   1st Test   TEST      Trent Bridge, Nottingham   04     AUGUST  15:30 IST
8   2nd Test   TEST                Lord's, London   12     AUGUST  15:30 IST
9   3rd Test   TEST             Headingley, Leeds   25     AUGUST  15:30 IST
10  4th Test   TEST              The Oval, London   02  SEPTEMBER  15:30 IST
11  5th Test   TEST      Old Trafford, Manchester   10  SEPTEMBER  15:30 IST

In [238]:
#Lets merge date and month column to view this in one column for better date visualization
cols=["date", "month"]
#Iterating through the data and month column and merging it seperated by comma
data["Date"]=data[cols].apply(lambda x: ', '.join(x.values.astype(str)), axis='columns')
data.drop(["date", "month"],axis=1, inplace=True) #Removing the initial date and month column
data["Date"] #Merged column

0          18, JUNE
1          13, JULY
2          16, JULY
3          18, JULY
4          21, JULY
5          23, JULY
6          25, JULY
7        04, AUGUST
8        12, AUGUST
9        25, AUGUST
10    02, SEPTEMBER
11    10, SEPTEMBER
Name: Date, dtype: object

In [239]:
data

Title Series                         Place       Time           Date
0      Final   TEST   The Ageas Bowl, Southampton  15:00 IST       18, JUNE
1    1st ODI    ODI  R Premadasa Stadium, Colombo  14:30 IST       13, JULY
2    2nd ODI    ODI  R Premadasa Stadium, Colombo  14:30 IST       16, JULY
3    3rd ODI    ODI  R Premadasa Stadium, Colombo  14:30 IST       18, JULY
4   1st T20I   T20I  R Premadasa Stadium, Colombo  19:00 IST       21, JULY
5   2nd T20I   T20I  R Premadasa Stadium, Colombo  19:00 IST       23, JULY
6   3rd T20I   T20I  R Premadasa Stadium, Colombo  19:00 IST       25, JULY
7   1st Test   TEST      Trent Bridge, Nottingham  15:30 IST     04, AUGUST
8   2nd Test   TEST                Lord's, London  15:30 IST     12, AUGUST
9   3rd Test   TEST             Headingley, Leeds  15:30 IST     25, AUGUST
10  4th Test   TEST              The Oval, London  15:30 IST  02, SEPTEMBER
11  5th Test   TEST      Old Trafford, Manchester  15:30 IST  10, SEPTEMBER

###### Assignment 4.3:
    Scrape the details of selenium exception from guru99.com.
Url = https://www.guru99.com/
You need to find following details:

A) Name
B) Description

Note: - From guru99 home page you have to reach to selenium exception handling page through code.


In [242]:
#Activating the Chrome Browser
driver=webdriver.Chrome('C:/Users/ak38/Desktop/FlipDataStore/chromedriver')

#Opening the URL
driver.get("https://www.guru99.com/")      

# to maximize the browser window
driver.maximize_window()

In [243]:
#Clicking on Selenium from Tutorials library
selenium=driver.find_element_by_xpath('//li/a[@title="Selenium"]').click()

In [244]:
#clicking on Exception handling tutorial to view all the exception name and description
exception_handle=driver.find_element_by_xpath('//a[@title="Selenium Exception Handling (Common Exceptions List)"]').click()

In [248]:
#creating 2 lists to store the data
exc_name=[]
exc_desc=[]

#Main element from which the table data is extracted
exceptions=driver.find_elements_by_xpath('//table[@class="table table-striped"]/tbody/tr')

#Iterating through the main element to get the required table details
for exception in exceptions:
    #Fetching the Exception Name
    exc_name.append(exception.find_element_by_xpath('td[1]').text) 
    #Fetching the Exception Description
    exc_desc.append(exception.find_element_by_xpath('td[2]').text)
len(exc_name), len(exc_desc)

(42, 42)

In [249]:
#Creating the dataframe
data=pd.DataFrame({})

#appending the lists to dectionary to dataframe
data['Exception_Name']=exc_name
data['Exception_Description']=exc_desc

#Viewing the final data
data

Exception_Name  \
0                        Exception name   
1            ElementNotVisibleException   
2         ElementNotSelectableException   
3                NoSuchElementException   
4                  NoSuchFrameException   
5               NoAlertPresentException   
6                 NoSuchWindowException   
7        StaleElementReferenceException   
8              SessionNotFoundException   
9                      TimeoutException   
10                   WebDriverException   
11            ConnectionClosedException   
12     ElementClickInterceptedException   
13      ElementNotInteractableException   
14             ErrorInResponseException   
15  ErrorHandler.UnknownServerException   
16         ImeActivationFailedException   
17             ImeNotAvailableException   
18         InsecureCertificateException   
19             InvalidArgumentException   
20         InvalidCookieDomainException   
21          InvalidCoordinatesException   
22          InvalidElementStateExceptio   
23            InvalidSessionIdException   
24       InvalidSwitchToTargetException   
25                  JavascriptException   
26                        JsonException   
27             NoSuchAttributeException   
28       MoveTargetOutOfBoundsException   
29               NoSuchContextException   
30                NoSuchCookieException   
31                    NotFoundException   
32          RemoteDriverServerException   
33                  ScreenshotException   
34           SessionNotCreatedException   
35           UnableToSetCookieException   
36           UnexpectedTagNameException   
37              UnhandledAlertException   
38      UnexpectedAlertPresentException   
39               UnknownMethodException   
40          UnreachableBrowserException   
41          UnsupportedCommandException   

                                Exception_Description  
0                                         Description  
1   This type of Selenium exception occurs when an...  
2   This Selenium exception occurs when an element...  
3   This Exception occurs if an element could not ...  
4   This Exception occurs if the frame target to b...  
5   This Exception occurs when you switch to no pr...  
6   This Exception occurs if the window target to ...  
7   This Selenium exception occurs happens when th...  
8   The WebDriver is acting after you quit the bro...  
9   Thrown when there is not enough time for a com...  
10  This Exception takes place when the WebDriver ...  
11  This type of Exception takes place when there ...  
12  The command may not be completed as the elemen...  
13  This Selenium exception is thrown when any ele...  
14  This happens while interacting with the Firefo...  
15  Exception is used as a placeholder in case if ...  
16  This expectation will occur when IME engine ac...  
17    It takes place when IME support is unavailable.  
18  Navigation made the user agent to hit a certif...  
19  It occurs when an argument does not belong to ...  
20  This happens when you try to add a cookie unde...  
21  This type of Exception matches an interacting ...  
22  It occurs when command can't be finished when ...  
23  This Exception took place when the given sessi...  
24  This occurs when the frame or window target to...  
25  This issue occurs while executing JavaScript g...  
26  It occurs when you afford to get the session w...  
27  This kind of Exception occurs when the attribu...  
28  It takes place if the target provided to the A...  
29           ContextAware does mobile device testing.  
30  This Exception occurs when no cookie matching ...  
31  This Exception is a subclass of WebDriverExcep...  
32  This Selenium exception is thrown when the ser...  
33            It is not possible to capture a screen.  
34  It happens when a new session could not be suc...  
35  This occurs if a driver is unable to set a coo...  
36  Happens if a support class did not get a web e...  
37  This expectation occurs when there is an al

###### Assignment 4.4:
     Scrape the details of State-wise GDP of India from statisticstime.com.
Url = http://statisticstimes.com/
You have to find following details:

A) Rank
B) State
C) GSDP(18-19)
D) GSDP(17-18)
E) Share(2017)
F) GDP($ billion)

Note: - From statisticstimes home page you have to reach to economy page through code.


In [264]:
#Activating the Chrome Browser
driver=webdriver.Chrome('C:/Users/ak38/Desktop/FlipDataStore/chromedriver')

#Opening the URL
driver.get("http://statisticstimes.com/")      

# to maximize the browser window
driver.maximize_window()

#click on the Economy tab to view the drop down menu
WebDriverWait(driver, 10).until(EC.element_to_be_clickable((By.XPATH, '//div[@class="navbar"]/div[2]/button'))).click()

#click on the option India from the drop down menu 
WebDriverWait(driver, 10).until(EC.element_to_be_clickable((By.XPATH, '//div[@class="navbar"]/div[2]/div/a[3]'))).click()

In [286]:
#Creating the lists to append the data points
gdprank=[]
state=[]
gsdp20=[]
gsdp19=[]
share19=[]
gdp=[]

#Main elements from which the data elements are extracted
tables=driver.find_elements_by_xpath('//div[@id="table_id_wrapper"]/table/tbody/tr')

#Iterating through the main element to extract the data points
for table in tables:
    #Fecthing the rank
    gdprank.append(table.find_element_by_xpath('td[@class="data1"][1]').text)
    #Fetching the state
    state.append(table.find_element_by_xpath('td[2]').text)
    #fecthing the GSDP(19-20)
    gsdp20.append(table.find_element_by_xpath('td[3]').text)
    #fetching the GSDP(19-18)
    gsdp19.append(table.find_element_by_xpath('td[4]').text)
    #fecthing the Share(18-19)
    share19.append(table.find_element_by_xpath('td[5]').text)
    #Fetching the GDP
    gdp.append(table.find_element_by_xpath('td[6]').text)
    
    
len(gdprank), len(state), len(gsdp20), len(gsdp19), len(share19), len(gdp)


(33, 33, 33, 33, 33, 33)

In [290]:
#Creating the dataframe
data=pd.DataFrame({})

#Appending each list to the dictionary 
data["Rank"]=gdprank
data["State"]=state
data["GSDP(19-20)"]=gsdp20
data["GSDP(18-19)"]=gsdp19
data["Share(2019)"]=share19
data["GDP($ billion)"]=gdp

#Viewing the final data
data

Rank                      State GSDP(19-20) GSDP(18-19) Share(2019)  \
0     1                Maharashtra           -   2,632,792      13.94%   
1     2                 Tamil Nadu   1,845,853   1,630,208       8.63%   
2     3              Uttar Pradesh   1,687,818   1,584,764       8.39%   
3     4                    Gujarat           -   1,502,899       7.96%   
4     5                  Karnataka   1,631,977   1,493,127       7.91%   
5     6                West Bengal   1,253,832   1,089,898       5.77%   
6     7                  Rajasthan   1,020,989     942,586       4.99%   
7     8             Andhra Pradesh     972,782     862,957       4.57%   
8     9                  Telangana     969,604     861,031       4.56%   
9    10             Madhya Pradesh     906,672     809,592       4.29%   
10   11                     Kerala           -     781,653       4.14%   
11   12                      Delhi     856,112     774,870       4.10%   
12   13                    Haryana     831,610     734,163       3.89%   
13   14                      Bihar     611,804     530,363       2.81%   
14   15                     Punjab     574,760     526,376       2.79%   
15   16                     Odisha     521,275     487,805       2.58%   
16   17                      Assam           -     315,881       1.67%   
17   18               Chhattisgarh     329,180     304,063       1.61%   
18   19                  Jharkhand     328,598     297,204       1.57%   
19   20                Uttarakhand           -     245,895       1.30%   
20   21            Jammu & Kashmir           -     155,956       0.83%   
21   22           Himachal Pradesh     165,472     153,845       0.81%   
22   23                        Goa      80,449      73,170       0.39%   
23   24                    Tripura      55,984      49,845       0.26%   
24   25                 Chandigarh           -      42,114       0.22%   
25   26                 Puducherry      38,253      34,433       0.18%   
26   27                  Meghalaya      36,572      33,481       0.18%   
27   28                     Sikkim      32,496      28,723       0.15%   
28   29                    Manipur      31,790      27,870       0.15%   
29   30                   Nagaland           -      27,283       0.14%   
30   31          Arunachal Pradesh           -      24,603       0.13%   
31   32                    Mizoram      26,503      22,287       0.12%   
32   33  Andaman & Nicobar Islands           -           -           -   

   GDP($ billion)  
0         399.921  
1         247.629  
2         240.726  
3         228.290  
4         226.806  
5         165.556  
6         143.179  
7         131.083  
8         130.791  
9         122.977  
10        118.733  
11        117.703  
12        111.519  
13         80.562  
14         79.957  
15         74.098  
16         47.982  
17         46.187  
18         45.145  
19         37.351  
20         23.690  
21         23.369  
22         11.115  
23          7.571  
24          6.397  
25          5.230  
26          5.086  
27          4.363  
28          4.233  
29          4.144  
30          3.737  
31          3.385  
32              -

###### Assignment 4.5:
    Scrape the details of trending repositories on Github.com.
Url = https://github.com/
You have to find the following details:

A) Repository title
B) Repository description
C) Contributors count
D) Language used

Note: - From the home page you have to click on the trending option from Explore menu through code

In [302]:
#Activating the Chrome Browser
driver=webdriver.Chrome('C:/Users/ak38/Desktop/FlipDataStore/chromedriver')

#Opening the URL
driver.get("https://github.com/")      
time.sleep(5)
# to maximize the browser window
driver.maximize_window()

In [304]:
#click on the Explore menu to view the drop down menu
WebDriverWait(driver, 10).until(EC.element_to_be_clickable((By.XPATH, '//ul[@class="d-lg-flex list-style-none"]/li[4]/details/summary'))).click()

#click on the option Trending from the drop down menu 
WebDriverWait(driver, 10).until(EC.element_to_be_clickable((By.XPATH, '//ul[@class="d-lg-flex list-style-none"]/li[4]/details/div/ul[2]/li[3]/a'))).click()

In [374]:
#Creating the lists to append the data points
rtitle=[]
rdesc=[]
contricount=[]
lang=[]

#Main elements from which the data elements are extracted
repos=driver.find_elements_by_xpath('//div[@class="Box"]/div[2]/article')

#Iterating through the main element to extract the data points
for repo in repos:
    #fetching the repository title
    rtitle.append(repo.find_element_by_xpath('h1/a').text)
    #fetching the repository description
    rdesc.append(repo.text.split("\n")[1])
    #fetching the count of the contributors
    contributer=repo.find_elements_by_xpath('div[2]/span[2]/a') #getting all the a tags 
    contricount.append(len(contributer)) #counting the a tags and appending to list one by one
    
    #Fetching the language details using try/catch blocks
    try:
        lang.append(repo.find_element_by_xpath('div[2]/span/span[2]').text)
    except NoSuchElementException:
        lang.append('-')

#Verifying if we have extracted the exact count by checking the length of the lists
len(rtitle), len(rdesc), len(contricount), len(lang)

(25, 25, 25, 25)

In [375]:
#Creating the dataframe
data=pd.DataFrame({})

#Appending each list to the dictionary 
data["Repo_Title"]=rtitle
data["Repo_Desc"]=rdesc
data["Contributors_count"]=contricount
data["Language"]=lang


#Viewing the final data
data

Repo_Title  \
0                          siduck76 / NvChad   
1                               star261 / jd   
2                             sqfmi / Watchy   
3                             jina-ai / jina   
4                          geohot / tinygrad   
5                     hotwired / turbo-rails   
6      jwasham / coding-interview-university   
7                         chriskiehl / Gooey   
8                     monyhar / monyhar-lite   
9                        chromium / chromium   
10  trimstray / the-book-of-secret-knowledge   
11          Asabeneh / 30-Days-Of-JavaScript   
12        lydiahallie / javascript-questions   
13                neon-age / Smart-Inspector   
14                         alibaba / formily   
15                 JDHelloWorld / jd_scripts   
16                 yunjey / pytorch-tutorial   
17                          vercel / next.js   
18                      programthink / books   
19                         ethersphere / bee   
20                        willmcgugan / rich   
21                              jdah / jdh-8   
22                        NVlabs / SegFormer   
23                       rustdesk / rustdesk   
24          adrianhajdin / portfolio_website   

                                            Repo_Desc  Contributors_count  \
0            beautiful neovim setup configured in lua                   5   
1                          JavaScript 166 96 Built by                   2   
2            Watchy - An Open Source E-Ink Smartwatch                   5   
3   An easier way to build neural search on the cloud                   5   
4   You like pytorch? You like micrograd? You love...                   5   
5                 Use Turbo in your Ruby on Rails app                   5   
6   A complete computer science study plan to beco...                   0   
7   Turn (almost) any Python command line program ...                   5   
8   梦弘浏览器 自主研发版本 - 完全自主研发，打破国外垄断，比 Chrome 快 600%。缺...                   1   
9   The official GitHub mirror of the Chromium source                   0   
10  A collection of inspiring lists, manuals, chea...                   0   
11  30 days of JavaScript programming challenge is...                   5   
12  A long list of (advanced) JavaScript questions...                   0   
13  Keeps your screen real-estate clean with refin...                   1   
14  Alibaba Group Unified Form Solution -- Support...                   5   
15                        JavaScript 311 125 Built by                   1   
16     PyTorch Tutorial for Deep Learning Researchers                   5   
17                                The React Framework                   5   
18                         【编程随想】收藏的电子书清单（多个学科，含下载链接）                   0   
19  Bee is a Swarm client implemented in Go. It’s ...                   5   
20  Rich is a Python library for rich text and bea...                   5   
21  An 8-bit minicomputer with a fully custom arch...                   1   
22  Official implementation of "SegFormer: Simple ...                   3   
23      An open source remote desktop client software                   5   
24  Tutorial created by Enyel Sequeira, taught by ...                   2   

      Language  
0          Lua  
1   JavaScript  
2            C  
3       Python  
4       Python  
5   JavaScript  
6            -  
7       Python  
8            C  
9            -  
10           -  
11  JavaScript  
12           -  
13          C#  
14  TypeScript  
15  JavaScript  
16      Python  
17  JavaScript  
18           -  
19          Go  
20      Python  
21           C  
22      Python  
23        Rust  
24  JavaScript

###### Assignment 4.6:
    Scrape the details of top 100 songs on billiboard.com.
Url = https://www.billboard.com/
You have to find the following details:

A) Song name
B) Artist name
C) Last week rank
D) Peak rank
E) Weeks on board

Note: - From the home page you have to click on the charts option then hot 100-page link through code.

In [392]:
#Activating the Chrome Browser
driver=webdriver.Chrome('C:/Users/ak38/Desktop/FlipDataStore/chromedriver')

#Opening the URL
driver.get("https://www.billboard.com/")      

# to maximize the browser window
driver.maximize_window()

#click on the Charts menu to view the drop down menu
WebDriverWait(driver, 10).until(EC.element_to_be_clickable((By.XPATH, '//div[@class="header__sticky__child"]/header/div/ul/li/a'))).click()

#click on the option view charts from The HOT 100 
WebDriverWait(driver, 10).until(EC.element_to_be_clickable((By.XPATH, '//div[@class="charts-landing__block"]/a/div[2]/div[2]/div'))).click()

In [416]:
#Creating the lists to append the data points
sname=[]
artist=[]
lastweekrank=[]
peakrank=[]
weeksonboard=[]

#Main elements from which the data elements are extracted
hot100=driver.find_elements_by_xpath('//div[@class="chart-list__wrapper"]/div/ol/li')

#Iterating through the main element to extract the data points
for hot in hot100:
    
    #Fetching the song name
    try:
        sname.append(hot.find_element_by_xpath('button/span[@class="chart-element__information"]/span').text)
    except NoSuchElementException:  #Exception to pass when there is adds coming in after 10 songs
        pass
    
    #fetching the Artist name
    try:
        artist.append(hot.find_element_by_xpath('button/span[@class="chart-element__information"]/span[2]').text)
    except NoSuchElementException:    #Exception to pass when there is adds coming in after 10 songs
        pass
    
    #Fetching the Last week rank
    try:
        lastweekrank.append(hot.find_element_by_xpath('button/div/div[@class="chart-element__meta text--center color--secondary text--last"]').text)
    except NoSuchElementException:    #Exception to pass when there is adds coming in after 10 songs
        pass
    
    #Fetching the Peak rank
    try:
        peakrank.append(hot.find_element_by_xpath('button/div/div[@class="chart-element__meta text--center color--secondary text--peak"]').text)
    except NoSuchElementException:    #Exception to pass when there is adds coming in after 10 songs
        pass
    
    #Fetching the Weeks on Chart
    try:
        weeksonboard.append(hot.find_element_by_xpath('button/div/div[@class="chart-element__meta text--center color--secondary text--week"]').text)
    except NoSuchElementException:    #Exception to pass when there is adds coming in after 10 songs
        pass
    
    
    
    
len(sname), len(artist), len(lastweekrank), len(peakrank), len(weeksonboard)


(100, 100, 100, 100, 100)

In [418]:
#Creating the dataframe
data=pd.DataFrame({})

#Appending each list to the dictionary 
data["Song_Name"]=sname
data["Artist_Name"]=artist
data["Last_Week_Rank"]=lastweekrank
data["Peak_Rank"]=peakrank
data["Weeks_on_Chart"]=weeksonboard


pd.set_option('display.max_rows', None)   #this code helps to view all the rows without hiding them

#Viewing the final data
data

Song_Name  \
0                              Butter   
1                            Good 4 U   
2                          Levitating   
3                             Peaches   
4                 Leave The Door Open   
5                     Save Your Tears   
6                        Kiss Me More   
7              Astronaut In The Ocean   
8                             Deja Vu   
9                            Yonaguni   
10     Montero (Call Me By Your Name)   
11                        Without You   
12                  Forever After All   
13                            Rapstar   
14                    Blinding Lights   
15                           Hats Off   
16                    Drivers License   
17                 Beautiful Mistakes   
18                            Traitor   
19                      Late At Night   
20                Voice Of The Heroes   
21                        Best Friend   
22             Heartbreak Anniversary   
23                         Heat Waves   
24                   Calling My Phone   
25                     Favorite Crime   
26                         Lost Cause   
27                            Happier   
28                                 Up   
29                               Mood   
30                               2040   
31                 Every Chance I Get   
32                          Telepatia   
33                       How It Feels   
34                             Brutal   
35                           Wockesha   
36                    Wants And Needs   
37                     Famous Friends   
38                                pov   
39                               Gone   
40                           Beat Box   
41                My Ex's Best Friend   
42                       Still Runnin   
43                         Track Star   
44            What You Know Bout Love   
45                         Who I Want   
46                            my.life   
47                      Back In Blood   
48                         Time Today   
49                            Lil Bit   
50                    Blame It On You   
51                             Nobody   
52                         Todo de Ti   
53                     Enough For You   
54                      Settling Down   
55                         Still Hood   
56                 Jealousy, Jealousy   
57                               Okay   
58                      Almost Maybes   
59                     Man Of My Word   
60                      Build A Bitch   
61   Breaking Up Was Easy In The 90's   
62                     Glad You Exist   
63                 pride.is.the.devil   
64       1 Step Forward, 3 Steps Back   
65                                Ski   
66                            Medical   
67                      Rich Off Pain   
68           Leave Before You Love Me   
69                              Lying   
70                         Snowflakes   
71              Single Saturday Night   
72                       That's Facts   
73                       Made For You   
74                         Your Power   
75                    No More Parties   
76                         We're Good   
77                       One Too Many   
78                             Please   
79                        Up The Side   
80                       Minimum Wage   
81                      Miss The Rage   
82                       Way Less Sad   
83                         Hope Ur OK   
84                          Quicksand   
85                             Arcade   
86                       Straightenin   
87                          Gang Gang   
88  Drinkin' Beer. Talkin' God. Amen.   
89                         Follow You   
90                  Chasing After You   
91                            Tell Em   
92                          4 Da Gang   
93                          Tombstone   
94                        What's Next   
95           Things A Man Oughta Know   
96                      Country Again   
97  Drunk 

###### Assignment 4.7:
     Scrape the details of Data science recruiters from naukri.com.
Url = https://www.naukri.com/
You have to find the following details:
    
A) Name
B) Designation
C) Company
D) Skills they hire for
E) Location

Note: - From naukri.com homepage click on the recruiters option and the on the search pane type Data science and 
click on search. All this should be done through code

In [14]:
#Activating the Chrome Browser
driver=webdriver.Chrome('C:/Users/ak38/Desktop/FlipDataStore/chromedriver')

#Opening the URL
driver.get("https://www.naukri.com/")      

# to maximize the browser window
driver.maximize_window()

#getting the URL of Recruiters option
recruiterURL=driver.find_element_by_xpath('//ul[@class="midSec menu"]/li[2]/a').get_attribute('href')
recruiterURL

#getting the recruters option URL and accessing it
driver.get(recruiterURL)
time.sleep(4)

#Locating the search bar by name
Job_search=driver.find_element_by_name("qp")  

#Clear the search bar
Job_search.clear()                                        

#enter the keyword
Job_search.send_keys("Data Science")   #Keyword

#Locating search button and click
Search_button=driver.find_element_by_id("qsbFormBtn")  
Search_button.click()       #clicking on search button

In [34]:
#Creating the lists to append the data points
name=[]
designation=[]
company=[]
skills=[]
location=[]

#Main elements from which the data elements are extracted
recruiters=driver.find_elements_by_xpath('//div[@class="outerRecSec"]/div')

#Iterating through the main element to extract the data points
for rec in recruiters:
    #Fetching the name
    name.append(rec.find_element_by_xpath('div/div[@class="vcard"]/p/a[1]/span[@class="fl ellipsis"]').text)
    #Fetching the designation
    designation.append(rec.find_element_by_xpath('div/div[@class="vcard"]/p/span[@class="ellipsis clr"]').text)
    #Fetching the Company
    company.append(rec.find_element_by_xpath('div/div[@class="vcard"]/p/a[2]/i[@class="cmpIcn"]').text)
    #Fetching the skills
    skills.append(rec.find_element_by_xpath('div/div[@class="hireSec highlightable"]').text)
    
    #Fetching the location
    try:
        location.append(rec.find_element_by_xpath('div/div[@class="vcard"]/p/span[2]/small').text)
    except NoSuchElementException:
        location.append('-')
    
#Verifying if all the entries are fetched
len(name), len(designation), len(company), len(skills), len(location)

(50, 50, 50, 50, 50)

In [35]:
#Creating the dataframe
data=pd.DataFrame({})

#Appending each list to the dictionary 
data["Name"]=name
data["Designation"]=designation
data["Company"]=company
data["Skills_They_Hire_for"]=skills
data["Location"]=location


pd.set_option('display.max_rows', None)   #this code helps to view all the rows without hiding them

#Viewing the final data
data

Name  \
0                                    Aakash Harit   
1                            shravan Kumar Gaddam   
2                        MARSIAN Technologies LLP   
3                                    Anik Agrawal   
4                                    subhas patel   
5    Abhishek - Only Analytics Hiring - India and   
6   Institute for Financial Management and Resear   
7                                     Balu Ramesh   
8                                   Asif Lucknowi   
9                                 InstaFinancials   
10                                Kalpana Dumpala   
11                                        Mubarak   
12                                 Kushal Rastogi   
13                                    Ruchi Dhote   
14                             Mahesh Babu Channa   
15                                   Kapil Devang   
16                                  Manisha Yadav   
17                                    Riya Rajesh   
18                           Rashmi Bhattacharjee   
19                                  Faizan Kareem   
20                                 Rithika dadwal   
21                                  Azahar Shaikh   
22                             Sandhya Khandagale   
23                                      Shaun Rao   
24                                          Manas   
25                                          kumar   
26                                   Sunil Vedula   
27                                    Rajat Kumar   
28                                    Priya Khare   
29                                Dhruv Dev Dubey   
30                                      Jayanth N   
31                              Radha Manivasagam   
32                                  Prateek Kumar   
33                                    Amit Sharma   
34                                          Kanan   
35                           Shashikant Chaudhary   
36                                           Brad   
37                                   Rutuja Pawar   
38                            Madhusudhan Sridhar   
39                                    Ankit Sinha   
40                                 Gaurav Chouhan   
41                                   Rashi Kacker   
42                                        Ashwini   
43                                   Balaji Kolli   
44                                 Rajani Nagaraj   
45                                    ROHIT Kumar   
46                                 Amir Chowdhury   
47                                 Shailja Mishra   
48                                   Sunny Sharma   
49                                     Sagar Lama   

                            Designation Company  \
0                            HR Manager           
1                     Company Recruiter           
2                            Company HR           
3                     Company Recruiter           
4                           Founder CEO           
5           Recruitment Lead Consultant           
6                     Programme Manager           
7                      HR Administrator           
8                              Director           
9                        Human Resource           
10                     Executive Hiring           
11                           Company HR           
12                           Company HR           
13  Senior Executive Talent Acquisition           
14                         HR Team Lead           
15                           HR Manager           
16                         HR Executive           
17           Manager Talent Acquisition           
18                              HR Head           
19                           HR MANAGER           
20                         HR Recruiter           
21                    Company Recruiter           
22                         HR Recruiter           
23              Manager Human Resources           
24              Lead Talent acquisition           
25             

###### Assignemt 4.8:        
    Scrape the details of Highest selling novels.
Url = https://www.theguardian.com/news/datablog/2012/aug/09/best-selling-books-all-time-fifty-shades-greycompare/

You have to find the following details:

A) Book name
B) Author name
C) Volumes sold
D) Publisher
E) Genre

In [2]:
#Activating the Chrome Browser
driver=webdriver.Chrome('C:/Users/ak38/Desktop/FlipDataStore/chromedriver')

#Opening the URL
driver.get("https://www.theguardian.com/news/datablog/2012/aug/09/best-selling-books-all-time-fifty-shades-grey-compare")      

# to maximize the browser window
driver.maximize_window()

In [6]:
#Creating the lists to append the data points
title=[]
author=[]
vsold=[]
publisher=[]
genre=[]

#Main elements from which the data elements are extracted
table=driver.find_elements_by_xpath('//div[@class="embed block"]/table/tbody/tr')

#Iterating through the main element to extract the data points
for tab in table:
    #Fetching the book Title
    title.append(tab.find_element_by_xpath('td[2]').text)
    #Fetching the Author Name
    author.append(tab.find_element_by_xpath('td[3]').text)
    #Fetching the volume sales
    vsold.append(tab.find_element_by_xpath('td[4]').text)
    #Fetching the Publisher data
    publisher.append(tab.find_element_by_xpath('td[5]').text)
    #Fetching the Genre
    genre.append(tab.find_element_by_xpath('td[6]').text)


#Verifying if all the entries are fetched
len(title), len(author), len(vsold), len(publisher), len(genre)

(100, 100, 100, 100, 100)

In [7]:
#Creating the dataframe
data=pd.DataFrame({})

#Appending each list to the dictionary 
data["Book_Name"]=title
data["Author_Name"]=author
data["Volumes_Sold"]=vsold
data["Publisher"]=publisher
data["Genre"]=genre


pd.set_option('display.max_rows', None)   #this code helps to view all the rows without hiding them

#Viewing the final data
data

Book_Name  \
0                                   Da Vinci Code,The   
1                Harry Potter and the Deathly Hallows   
2            Harry Potter and the Philosopher's Stone   
3           Harry Potter and the Order of the Phoenix   
4                                Fifty Shades of Grey   
5                 Harry Potter and the Goblet of Fire   
6             Harry Potter and the Chamber of Secrets   
7            Harry Potter and the Prisoner of Azkaban   
8                                   Angels and Demons   
9   Harry Potter and the Half-blood Prince:Childre...   
10                                Fifty Shades Darker   
11                                           Twilight   
12  Girl with the Dragon Tattoo,The:Millennium Tri...   
13                                 Fifty Shades Freed   
14                                    Lost Symbol,The   
15                                           New Moon   
16                                    Deception Point   
17                                            Eclipse   
18                                   Lovely Bones,The   
19  Curious Incident of the Dog in the Night-time,The   
20                                   Digital Fortress   
21               Short History of Nearly Everything,A   
22   Girl Who Played with Fire,The:Millennium Trilogy   
23                                      Breaking Dawn   
24  Very Hungry Caterpillar,The:The Very Hungry Ca...   
25                                       Gruffalo,The   
26                            Jamie's 30-Minute Meals   
27                                    Kite Runner,The   
28                                            One Day   
29                           Thousand Splendid Suns,A   
30  Girl Who Kicked the Hornets' Nest,The:Millenni...   
31                           Time Traveler's Wife,The   
32                                          Atonement   
33                      Bridget Jones's Diary:A Novel   
34                    World According to Clarkson,The   
35                         Captain Corelli's Mandolin   
36                              Sound of Laughter,The   
37                                         Life of Pi   
38                                     Billy Connolly   
39                                  Child Called It,A   
40                               Gruffalo's Child,The   
41             Angela's Ashes:A Memoir of a Childhood   
42                                           Birdsong   
43              Northern Lights:His Dark Materials S.   
44                                          Labyrinth   
45             Harry Potter and the Half-blood Prince   
46                                           Help,The   
47                                        Man and Boy   
48                                Memoirs of a Geisha   
49  No.1 Ladies' Detective Agency,The:No.1 Ladies'...   
50                                         Island,The   
51                                     PS, I Love You   
52  You are What You Eat:The Plan That Will Change...   
53                             Shadow of the Wind,The   
54                       Tales of Beedle the Bard,The   
55                                         Broker,The   
56  Dr. Atkins' New Diet Revolution:The No-hunger,...   
57             Subtle Knife,The:His Dark Materials S.   
58  Eats, Shoots and Leaves:The Zero Tolerance App...   
59                         Delia's How to Cook:(Bk.1)   
60                                           Chocolat   
61                     Boy in the Striped Pyjamas,The   
62                                 My Sister's Keeper   
63           Amber Spyglass,The:His Dark Materials S.   
64                              To Kill a Mockingbird   
65  Men are from Mars, Women are from Venus:A Prac...   
66                                         Dear Fatty   
67           Short History of Tractors in Ukrainian,A   
68                                           Hannibal   
69                              Lord of the Rings,The 

###### Assignment 4.9:
    Scrape the details most watched tv series of all time from imdb.com.
 Url = https://www.imdb.com/list/ls095964455/
    
You have to find the following details:
    
A) Name
B) Year span
C) Genre
D) Run time
E) Ratings
F) Votes

In [38]:
#Activating the Chrome Browser
driver=webdriver.Chrome('C:/Users/ak38/Desktop/FlipDataStore/chromedriver')

#Opening the URL
driver.get("https://www.imdb.com/list/ls095964455/")      

# to maximize the browser window
driver.maximize_window()

In [43]:
#Creating the lists to append the data points
name=[]
yearspan=[]
genre=[]
runtime=[]
ratings=[]
votes=[]

#Main elements from which the data elements are extracted
shows=driver.find_elements_by_xpath('//div[@class="lister list detail sub-list"]/div[3]/div')

#Iterating through the main element to extract the data points
for show in shows:
    #Fetching the name 
    name.append(show.find_element_by_xpath('div[2]/h3/a').text)
    #Fetching the Year Span
    yearspan.append(show.find_element_by_xpath('div[2]/h3/span[2]').text)
    #Fetching the genre
    genre.append(show.find_element_by_xpath('div[2]/p/span[@class="genre"]').text)
    #Fetching the Run Time
    runtime.append(show.find_element_by_xpath('div[2]/p/span[@class="runtime"]').text)
    #Fetching the Rating
    ratings.append(show.find_element_by_xpath('div[2]/div/div/span[@class="ipl-rating-star__rating"]').text)
    #Fetching the Votes
    votes.append(show.find_element_by_xpath('div[2]/p/span[@name="nv"]').text)

#Verifying if all the entries are fetched
len(name), len(yearspan), len(genre), len(runtime), len(ratings), len(votes)

(100, 100, 100, 100, 100, 100)

In [44]:
#Creating the dataframe
data=pd.DataFrame({})

#Appending each list to the dictionary 
data["Name"]=name
data["Yean_Span"]=yearspan
data["Genre"]=genre
data["Run_Time"]=runtime
data["Ratings"]=ratings
data["Votes"]=votes


pd.set_option('display.max_rows', None)   #this code helps to view all the rows without hiding them

#Viewing the final data
data

Name    Yean_Span                         Genre  \
0                  Game of Thrones  (2011–2019)      Action, Adventure, Drama   
1                  Stranger Things     (2016– )        Drama, Fantasy, Horror   
2                 The Walking Dead  (2010–2022)       Drama, Horror, Thriller   
3                   13 Reasons Why  (2017–2020)      Drama, Mystery, Thriller   
4                          The 100  (2014–2020)        Drama, Mystery, Sci-Fi   
5          Orange Is the New Black  (2013–2019)          Comedy, Crime, Drama   
6                        Riverdale     (2017– )         Crime, Drama, Mystery   
7                   Grey's Anatomy     (2005– )                Drama, Romance   
8                        The Flash     (2014– )      Action, Adventure, Drama   
9                            Arrow  (2012–2020)      Action, Adventure, Crime   
10                     Money Heist  (2017–2021)        Action, Crime, Mystery   
11             The Big Bang Theory  (2007–2019)               Comedy, Romance   
12                    Black Mirror     (2011– )       Drama, Sci-Fi, Thriller   
13                        Sherlock  (2010–2017)         Crime, Drama, Mystery   
14                         Vikings  (2013–2020)      Action, Adventure, Drama   
15             Pretty Little Liars  (2010–2017)       Drama, Mystery, Romance   
16             The Vampire Diaries  (2009–2017)        Drama, Fantasy, Horror   
17           American Horror Story     (2011– )       Drama, Horror, Thriller   
18                    Breaking Bad  (2008–2013)        Crime, Drama, Thriller   
19                         Lucifer     (2016– )         Crime, Drama, Fantasy   
20                    Supernatural  (2005–2020)        Drama, Fantasy, Horror   
21                    Prison Break  (2005–2017)          Action, Crime, Drama   
22     How to Get Away with Murder  (2014–2020)         Crime, Drama, Mystery   
23                       Teen Wolf  (2011–2017)        Action, Drama, Fantasy   
24                    The Simpsons     (1989– )             Animation, Comedy   
25                Once Upon a Time  (2011–2018)   Adventure, Fantasy, Romance   
26                          Narcos  (2015–2017)       Biography, Crime, Drama   
27                       Daredevil  (2015–2018)          Action, Crime, Drama   
28                         Friends  (1994–2004)               Comedy, Romance   
29           How I Met Your Mother  (2005–2014)               Comedy, Romance   
30                           Suits  (2011–2019)                 Comedy, Drama   
31                       Mr. Robot  (2015–2019)        Crime, Drama, Thriller   
32                   The Originals  (2013–2018)        Drama, Fantasy, Horror   
33                       Supergirl  (2015–2021)      Action, Adventure, Drama   
34                     Gossip Girl  (2007–2012)                Drama, Romance   
35                          Sense8  (2015–2018)        Drama, Mystery, Sci-Fi   
36                          Gotham  (2014–2019)          Action, Crime, Drama   
37                       Westworld     (2016– )        Drama, Mystery, Sci-Fi   
38                   Jessica Jones  (2015–2019)          Action, Crime, Drama   
39                   Modern Family  (2009–2020)        Comedy, Drama, Romance   
40                  Rick and Morty     (2013– )  Animation, Adventure, Comedy   
41                   Shadowhunters  (2016–2019)        Action, Drama, Fantasy   
42    The End of the F***ing World  (2017–2019)      Adventure, Comedy, Crime   
43                  House of Cards  (2013–2018)                         Drama   
44                            Dark  (2017–2020)         Crime, Drama, Mystery   
45                           Elite     (2018– )        Crime, Drama, Thriller   
46                   Sex Education     (2019– )                 Comedy, Drama   
47                       Shameless  (2011–2021)                 Comedy, Drama   
48                        New Girl  (2011–2018)              

###### Assignment 4.10:
    Details of Datasets from UCI machine learning repositories.
 Url = https://archive.ics.uci.edu/
    
 You have to find the following details:
    
A) Dataset name
B) Data type
C) Task
D) Attribute type
E) No of instances
F) No of attribute
G) Year

Note: - from the home page you have to go to the Show All Dataset page through code

In [72]:
#Activating the Chrome Browser
driver=webdriver.Chrome('C:/Users/ak38/Desktop/FlipDataStore/chromedriver')

#Opening the URL
driver.get("https://archive.ics.uci.edu/")      

# to maximize the browser window
driver.maximize_window()

#Clicking on the View ALL Data Sets option to view all the Data sets
View_all_Datasets=driver.find_element_by_xpath('//span[@class="whitetext"]/a/font/b').click()

In [46]:
#Clicking on the View ALL Data Sets option to view all the Data sets
View_all_Datasets=driver.find_element_by_xpath('//span[@class="whitetext"]/a/font/b').click()

In [73]:
#Creating the lists to append the data points
ds_name=[]
dtype=[]
task=[]
attr_type=[]
instances=[]
attributes=[]
year=[]

#Main elements from which the data elements are extracted
table=driver.find_elements_by_xpath('//table/tbody/tr/td[2]/table[2]/tbody/tr')

#Iterating through the main element to extract the data points
for tab in table:
    #Fetching the data set name
    try:
        ds_name.append(tab.find_element_by_xpath('td[1]/table/tbody/tr/td[2]/p/b').text)
    except NoSuchElementException:
        ds_name.append('-')
    
    #Fetching the data type
    try:
        dtype.append(tab.find_element_by_xpath('td[2]/p').text)
    except NoSuchElementException:
        dtype.append('-')
    
    #Fetching the task
    try:
        task.append(tab.find_element_by_xpath('td[3]/p').text)
    except NoSuchElementException:
        task.append('-')
        
    try:
        attr_type.append(tab.find_element_by_xpath('td[4]/p').text)
    except NoSuchElementException:
        attr_type.append('-')
        
    try:
        instances.append(tab.find_element_by_xpath('td[5]/p').text)
    except NoSuchElementException:
        instances.append('-')
        
    try:
        attributes.append(tab.find_element_by_xpath('td[6]/p').text)
    except NoSuchElementException:
        attributes.append('-')
        
    try:
        year.append(tab.find_element_by_xpath('td[7]/p').text)
    except NoSuchElementException:
        year.append('-')

#Verifying if all the entries are fetched
len(ds_name), len(dtype), len(task), len(attr_type), len(instances), len(attributes), len(year)

(589, 589, 589, 589, 589, 589, 589)

In [74]:
#Creating the dataframe
data=pd.DataFrame({})

#Appending each list to the dictionary 
data["DataSet_Name"]=ds_name
data["Data_Type"]=dtype
data["Task"]=task
data["Attribute_Type"]=attr_type
data["Instances"]=instances
data["Attributes"]=attributes
data["Year"]=year


pd.set_option('display.max_rows', None)   #this code helps to view all the rows without hiding them

#Viewing the final data
data

DataSet_Name  \
0                                                    -   
1                                              Abalone   
2                                                Adult   
3                                            Annealing   
4                         Anonymous Microsoft Web Data   
5                                           Arrhythmia   
6                                Artificial Characters   
7                                 Audiology (Original)   
8                             Audiology (Standardized)   
9                                             Auto MPG   
10                                          Automobile   
11                                              Badges   
12                                       Balance Scale   
13                                            Balloons   
14                                       Breast Cancer   
15                  Breast Cancer Wisconsin (Original)   
16                Breast Cancer Wisconsin (Prognostic)   
17                Breast Cancer Wisconsin (Diagnostic)   
18                                  Pittsburgh Bridges   
19                                      Car Evaluation   
20                                       Census Income   
21                   Chess (King-Rook vs. King-Knight)   
22                     Chess (King-Rook vs. King-Pawn)   
23                          Chess (King-Rook vs. King)   
24                             Chess (Domain Theories)   
25                                       Bach Chorales   
26                                           Connect-4   
27                                     Credit Approval   
28                           Japanese Credit Screening   
29                                   Computer Hardware   
30                         Contraceptive Method Choice   
31                                           Covertype   
32                                      Cylinder Bands   
33                                         Dermatology   
34                                            Diabetes   
35           DGP2 - The Second Data Generation Program   
36                              Document Understanding   
37                                 EBL Domain Theories   
38                                      Echocardiogram   
39                                               Ecoli   
40                                               Flags   
41                                    Function Finding   
42                                Glass Identification   
43                                 Haberman's Survival   
44                                          Hayes-Roth   
45                                       Heart Disease   
46                                           Hepatitis   
47                                         Horse Colic   
48                                                 ICU   
49                                  Image Segmentation   
50                             Internet Advertisements   
51                                          Ionosphere   
52                                                Iris   
53                                              ISOLET   
54                                             Kinship   
55                                     Labor Relations   
56                                  LED Display Domain   
57                                              Lenses   
58                                  Letter Recognition   
59                                     Liver Disorders   
60                                      Logic Theorist   
61                                         Lung Cancer   
62                                        Lymphography   
63                                 Mechanical Analysis   
64                                           Meta-data   
65                                       Mobile Robots   
66         Molecular Biology (Promoter Gene Sequences)   
67     Molecular Biology (Protein Secondary Structure)   
68   Molecular Biology (Splice-junction 